micropip/package_index.py

```diff
@@ -313,14 +313,9 @@ async def query_package(
             logger.debug("Url has no placeholder, appending package name : %r", url)
         try:
             metadata, headers = await fetch_string_and_headers(url, _fetch_kwargs)
-        except HttpStatusError as e:
-            if e.status_code == 404:
-                logger.debug("NotFound (404) for %r, trying next index.", url)
-                continue
-            logger.debug(
-                "Error fetching %r (%s), trying next index.", url, e.status_code
-            )
-            raise
+        except Exception as e:
+            logger.debug("Error fetching metadata for %r: %r", url, e)
```

Comment:
Maybe let's make the log message more detailed.

```suggestion
            logger.debug("Error fetching metadata for the package %r from (%r): %r, trying next index.", name, url, e)
```